src/main/kotlin/interactor/GetAvgFruitNVeggieInteractor.kt

```diff
@@ -0,0 +1,49 @@
+package interactor
+
+import model.CityEntity
+import model.FruitAndVegetablesPrices
+class GetAvgFruitNVeggieInteractor(
+    private val dataSource: CostOfLivingDataSource,
+) {
+    fun execute(): List<CityEntity>? {
+        val list = dataSource.getAllCitiesData()
+        if (list.isEmpty()) return null
+        return dataSource.getAllCitiesData()
+
+            .filter(::excludeNullSalaries)
+            .filter(::excludeNullFruitsVegPrices)
+            .sortedBy {
+                FruitsAndVegetablesPrice(it.fruitAndVegetablesPrices, it.averageMonthlyNetSalaryAfterTax!!)
+            }
+            .take(10)
+    }
+
+
+
+    private fun excludeNullSalaries(city: CityEntity): Boolean {
+        return city.averageMonthlyNetSalaryAfterTax != null
+    }
+
+    private fun excludeNullFruitsVegPrices(city: CityEntity): Boolean {
+        with(city.fruitAndVegetablesPrices) {
+            return apples1kg != null &&
+                    banana1kg != null &&
+                    oranges1kg != null &&
+                    tomato1kg != null &&
+                    potato1kg != null &&
+                    onion1kg != null &&
+                    lettuceOneHead != null
+        }
+    }
+        fun FruitsAndVegetablesPrice(
```

Comment:
see [coding conventions](https://kotlinlang.org/docs/coding-conventions.html#function-names)
 > Names of functions, properties and local variables start with a lowercase letter

```suggestion
        fun fruitsAndVegetablesPrice(
```